### PR TITLE
Dropped dependency to nipy

### LINKED ISSUE #2231: Replace nipy dependency with nibabel for i/o
for `resample.resample_nipy`, we can use nibabel objects (more commonly used in SCT) instead of nipy ones, see Example 44: https://www.programcreek.com/python/example/98176/nibabel.load

### LINKED ISSUE #2267: ImportError: cannot import name 'factorial' from 'scipy.misc'
New deprecation from nipy: https://github.com/nipy/nipy/issues/453

more info at: https://scipy.github.io/devdocs/release.1.3.0.html#scipy-interpolate-changes
Travis fail: https://travis-ci.org/neuropoly/spinalcordtoolbox/jobs/535398407#L1012

### LINKED ISSUE #2364: Drop dependency to nipy
Arguments for dropping nipy:
- We are having several issues with this package: #2362, #2267, #2231
- nipy is no more maintained. See: https://github.com/nipy/nipy/pull/454
- We have a small dependency: only `sct_resample` will need to be refactored.

### LINKED ISSUE #2399: Installation failing on 4.0.0
Likely related to Python3 (since 4.0.0_beta.2 is working and is relying on Py2k).
More info at: http://forum.spinalcordmri.org/t/sct-4-0-0-install-failed-failed-to-build-wheel-for-nipy/138

Note that the user is using .zshrc, but I don't think this is related to the issue.

@gmotzespina can you please have a look at it?

### PR DESCRIPTION
**Pull request by jcohenadad**
Because the fix proposed in #2269 (use a fork to nipy) does not solve all issues related to nipy, this PR proposes to replace nipy's resampling method by nibabel's resampling method, thereby dropping dependency to nipy.

Details of the implementation:
- Replaced `resampling.resample_nipy` by `resampling.resample_nib`
- Removed nipy from requirements

Fixes #2267, Fixes #2231, Fixes #2399, Fixes #2364

### CODE REVIEW

**Comment by jcohenadad** [comment]
Investigated the approach of having the fork on pypi, but it seems to be a [bad idea](https://www.reddit.com/r/learnpython/comments/2zclm9/fork_on_pypi/)

**Comment by jcohenadad** [comment]
Exploring the possibility to use [nilearn](https://nilearn.github.io/modules/generated/nilearn.image.resample_img.html) or [nibabel](https://nipy.org/nibabel/reference/nibabel.processing.html#resample-to-output) in place or nipy's resample function.

**Comment by jcohenadad** [comment]
Results of batch_processing:

git-jca/2364-nipy 7b639e5b2a3b1f7f9e3044b10a93c9c55d094a58
~~~
t2/CSA:          74.01878410546755
mt/MTR(WM):      54.27155182125294
t2s/CSA_GM:      12.771865152681794
t2s/CSA_WM:      64.85976674551333
dmri/FA(CST_r):  0.7919877164729336
dmri/FA(CST_l):  0.770509522891699
~~~

4.0.0
~~~
t2/CSA:          74.05757720942549
mt/MTR(WM):      54.29907016962242
t2s/CSA_GM:      12.774964384249389
t2s/CSA_WM:      64.64670129025727
dmri/FA(CST_r):  0.7723221487784772
dmri/FA(CST_l):  0.746995304331143
~~~

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-numpy # need to install first otherwise dependency issues
+numpy  # need to install first otherwise dependency issues
```
Comment (gmotzespina) [inline requirements.txt]: Is this extra space needed?

File: spinalcordtoolbox/resampling.py [hunk 2]
```diff
@@ -9,37 +9,37 @@
...
 
 from __future__ import division, absolute_import
 
 import logging
-import nipy
 import numpy as np
-from nipy.algorithms.registration.resample import resample as n_resample
+import nibabel as nib
+from nibabel.processing import resample_from_to
+
```
Comment (gmotzespina) [inline spinalcordtoolbox/resampling.py]: Maybe remove this extra line break?

Comment (jcohenadad) [reply spinalcordtoolbox/resampling.py]: blank line is to separate import groups, as per [PEP8 convention](https://www.python.org/dev/peps/pep-0008/#imports)

File: requirements.txt (hunk 1 above)
Comment (jcohenadad) [reply requirements.txt]: [flake8 rule](https://lintlyci.github.io/Flake8Rules/rules/E261.html)

**Pull request merged**